torch/nn/init.py

```diff
@@ -661,11 +677,11 @@ def sparse_(
 
 
 # for backward compatibility
-def _make_deprecate(meth):
+def _make_deprecate(meth: Callable[..., Tensor]) -> Callable[..., Tensor]:
     new_name = meth.__name__
     old_name = new_name[:-1]
 
-    def deprecated_init(*args, **kwargs):
+    def deprecated_init(*args: tuple[Any], **kwargs: dict[str, Any]) -> Tensor:
```

Comment:
```suggestion
def _make_deprecate(meth: Callable[_P, _R]) -> Callable[_P, _R]:
    new_name = meth.__name__
    old_name = new_name[:-1]

    def deprecated_init(*args : _P.args, **kwargs: _P.kwargs):
```
Where _R is a typing.TypeVar and _P is a typing_extensions.ParamSpec

Reply:
Besides, this should all be replaced with typing_extensions.deprecated anyway